pkg/limits/ingest_limits.go

```diff
@@ -376,29 +392,33 @@ func (s *IngestLimits) ExceedsLimits(_ context.Context, req *proto.ExceedsLimits
 			continue
 		}
 
-		streams[partitionID] = append(streams[partitionID], Stream{
-			Hash:       stream.StreamHash,
-			LastSeenAt: recordTime,
-			TotalSize:  stream.TotalSize,
-		})
+		streams = append(streams, stream)
 	}
 
-	storeRes := make(map[Reason][]uint64)
-	cond := streamLimitExceeded(maxActiveStreams, storeRes)
+	rejected := make(map[uint64]Reason)
```

Comment:
I'm confused why you chose to return the accepted streams as a return value for `StoreCond`, but return the rejected streams by capturing the values in a closure? Is there a reason for that, because I would just return both from `StoreCond`?